src/Sandbox.php

```diff
@@ -38,87 +39,55 @@ class Sandbox
     private ?int $expirationAfterHours = null;
     private ?string $deletedTimestamp = null;
 
-    public function __construct(?array $sandbox = null)
-    {
-        if (!empty($sandbox['id'])) {
-            $this->setId((string) $sandbox['id']);
-        }
-        if (!empty($sandbox['projectId'])) {
-            $this->setProjectId((string) $sandbox['projectId']);
-        }
-        if (!empty($sandbox['tokenId'])) {
-            $this->setTokenId((string) $sandbox['tokenId']);
-        }
-        if (!empty($sandbox['configurationId'])) {
-            $this->setConfigurationId((string) $sandbox['configurationId']);
-        }
-        if (!empty($sandbox['physicalId'])) {
-            $this->setPhysicalId($sandbox['physicalId']);
-        }
 
-        if (!empty($sandbox['type'])) {
-            $this->setType($sandbox['type']);
+    public static function fromArray(array $in): self
+    {
+        $sandbox = new Sandbox();
+        if (!isset($in['id'])) {
+            throw new InvalidApiResponseException('Property id is missing from API response');
         }
-        if (!empty($sandbox['size'])) {
-            $this->setSize($sandbox['size']);
+        $sandbox->setId((string) $in['id']);
+        if (!isset($in['projectId'])) {
+            throw new InvalidApiResponseException('Property projectId is missing from API response');
         }
-
-        if (!empty($sandbox['user'])) {
-            $this->setUser($sandbox['user']);
+        $sandbox->setProjectId((string) $in['projectId']);
+        if (!isset($in['tokenId'])) {
+            throw new InvalidApiResponseException('Property tokenId is missing from API response');
         }
-        if (!empty($sandbox['password'])) {
-            $this->setPassword($sandbox['password']);
+        $sandbox->setTokenId((string) $in['tokenId']);
+        if (!isset($in['configurationId'])) {
```

Comment:
jeste bych timpadem vyhodil ten null u deklarace
private ?string $configurationId = null;

a myslim, ze je zbytecny u tech deklaraci mit to `= null` ne?

Reply:
jj.